config/config.go

```diff
@@ -418,6 +418,26 @@ type Local struct {
 	// features like catchpoint catchup would be rendered completly non-operational, and many of the node inner
 	// working would be completly dis-functional.
 	DisableNetworking bool `version[16]:"false"`
+
+	// ForceFetchTransactions allows to explicitly configure a node to retrieve all the transactions
+	// into it's transaction pool, even if those would not be required as the node doesn't
+	// participate in the consensus or used to relay transactions.
+	ForceFetchTransactions bool `version[17]:"false"`
```

Comment:
This help to maintain the existing behavior, which is used ( incorrectly ) for load balancers. On load balancers setup, a transaction is being sent to any of the nodes, and the being checked if it's in the transaction pool. The second check might end up on a different machine. The issue is that it's not guaranteed to work correctly ( it was never working correctly, btw ), since the transaction propagation is not an atomic operation.

Reply:
I don't think we should provide this option. The current behavior is already "broken" and users must account for the case when a transaction has not yet propagated to other nodes. The txsync changes make the situation a little worse, but it's still racy and shouldn't be encouraged by providing an option like this.

The developer site provides a "wait for confirmation" algorithm that works regardless of this setting by waiting for the tx to appear in a block: https://developer.algorand.org/docs/build-apps/hello_world/#wait-for-confirmation

Reply:
I agree that the current behavior is broken. I hope that users would only use this option as a temporary solution, until they implement the functionality correctly. Unfortunately, without that, we're guaranteed to fail in certain cases ( at least for now ), and I'd like to prevent that from happening.